src/components/ButtonPanel.js

```diff
@@ -1,40 +1,45 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import Button from './Button';
 
-function ButtonPanel() {
+function ButtonPanel({ clickHandler }) {
   return (
     <div className="button-panel">
       <div className="group">
-        <Button name="AC" color="#dfdfdf" />
-        <Button name="+/-" color="#dfdfdf" />
-        <Button name="%" color="#dfdfdf" />
-        <Button name="÷" />
+        <Button name="AC" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="+/-" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="%" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="÷" clickHandler={clickHandler} />
       </div>
       <div className="group">
-        <Button name="7" color="#dfdfdf" />
-        <Button name="8" color="#dfdfdf" />
-        <Button name="9" color="#dfdfdf" />
-        <Button name="x" />
+        <Button name="7" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="8" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="9" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="x" clickHandler={clickHandler} />
       </div>
       <div className="group">
-        <Button name="4" color="#dfdfdf" />
-        <Button name="5" color="#dfdfdf" />
-        <Button name="6" color="#dfdfdf" />
-        <Button name="-" />
+        <Button name="4" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="5" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="6" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="-" clickHandler={clickHandler} />
       </div>
       <div className="group">
-        <Button name="1" color="#dfdfdf" />
-        <Button name="2" color="#dfdfdf" />
-        <Button name="3" color="#dfdfdf" />
-        <Button name="+" />
+        <Button name="1" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="2" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="3" color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="+" clickHandler={clickHandler} />
       </div>
       <div className="group">
-        <Button name="0" color="#dfdfdf" wide />
-        <Button name="." color="#dfdfdf" />
-        <Button name="=" />
+        <Button name="0" color="#dfdfdf" clickHandler={clickHandler} wide />
+        <Button name="." color="#dfdfdf" clickHandler={clickHandler} />
+        <Button name="=" clickHandler={clickHandler} />
       </div>
```

Comment:
**OPTIONAL**: A lot of duplicated code 🤔, do you think we could do this without repeating ourselves so much?

Reply:
Yes, I can create a function that returns the button component and pass name and color as arguments.